root/etc/systemd/system/discovery-setup.path

```diff
@@ -0,0 +1,11 @@
+[Unit]
+# Network-Manager-wait-online doesn't seem to wait *quite* long enough
+# for the custom post-up command to write the next-server environment
+# file. So we use a path target to wait for it to be written.
+Description=Waiting for ZIP_SERVER environment file to be created
+
+[Path]
+PathExists=/etc/default/discovery-zip-server
+
+[Install]
+WantedBy=multi-user.target
```

Comment:
Yeah, wait-online didn;t wait *quite* long enough. I was about to add a sleep when I saw that you could do this instead :)